src/atomicgen.py

```diff
@@ -0,0 +1,130 @@
+        if m is None:
+            continue
+            
+        nuc = (10000000 * int(m.group(1))) + (10000 * int(m.group(2)))
+        error = 1E-6 * float(m.group(5).strip().replace('#', ''))
+        atomic_mass_error = "  atomic_mass_error["+str(nuc)+"] = "+str(error)+";\n"
+
+    return atomic_mass_error
+
+# print the masses map
+def print_atomic_mass():
+    atomic_mass = ""
+    file = open('../pyne/dbgen/mass.mas12','r')
+    amdc_regex = re.compile('[ \d-]*? (\d{1,3})[ ]{1,4}(\d{1,3}) [A-Z][a-z]? .*? (\\d{1,3}) ([ #.\d]{5,12}) ([ #.\d]+)[ ]*?$')
+    for line in file:
+        m = amdc_regex.search(line)
+        if m is None:
+            continue
+            
+        nuc = (10000000 * int(m.group(1))) + (10000 * int(m.group(2)))
+        mass = float(m.group(3)) + 1E-6 * float(m.group(4).strip().replace('#',''))
+        atomic_mass += "  atomic_mass_map["+str(nuc)+"] = "+str(mass)+";\n"
+
+    return atomic_mass
+
+# print the abundances map
+def print_abundances():
+    isotopic_abundances = ""
+    file = open('../pyne/dbgen/abundances.txt','r')
+
+    for line in file:
+        # tokenise the line
+        splitted = line.split(' ')
+        tokens = []
+        for item in splitted:
+            if item != '':
+                tokens.append(item)
+                # ignore the comment line
+        if tokens[0] != "#":
+            tokens[3] = tokens[3].replace("\n","")
+            isotopic_abundances += "  natural_abund_map["+str((int(tokens[0])*10000000)+(int(tokens[2])*10000))+"] = "+str(tokens[3])+";\n"
+    return isotopic_abundances
+
+def print_cpp_file(filename):
+    #empty string
+    cpp_file = ""
+    cpp_file += "// Implements basic nuclear data functions.\n"
+    cpp_file += "#ifndef PYNE_IS_AMALGAMATED\n"
+    cpp_file += "#include \"atomic_data.h\"\n"
+    cpp_file += "#endif\n"
+    cpp_file += "  \n"
+    cpp_file += "void pyne::_load_atomic_mass_map_memory() { \n"
+    cpp_file += "  // header version of atomic weight table data \n"
+    cpp_file += "  //see if the data table is already loaded\n"
+    cpp_file += "  if(!atomic_mass_map.empty()) {\n"
+    cpp_file += "    return;\n"
+    cpp_file += "  } else { \n"
+    cpp_file += "    _insert_atomic_mass_map();\n"
+    cpp_file += "  }\n"
+    cpp_file += "  //see if the data table is already loaded\n"
+    cpp_file += "  if(!natural_abund_map.empty()) {\n"
+    cpp_file += "    return;\n"
+    cpp_file += "  } else { \n"
+    cpp_file += "    _insert_abund_map();\n"
+    cpp_file += "  }\n"
+    cpp_file += "}\n"
+    cpp_file += "\n"
+    cpp_file += "void pyne::_insert_atomic_mass_map() { \n"
+    cpp_file += print_atomic_mass()
+    cpp_file += "}\n"
+    cpp_file += "\n"
+    cpp_file += "void pyne::_insert_abund_map() { \n"
+    cpp_file += print_abundances()
+    cpp_file += "}\n"
+
+    f = open(filename,'w')
+    f.write(cpp_file)
+    f.close()
+
+print_cpp_file("atomic_data.cpp")
+print_header_file("atomic_data.h")
```

Comment:
These lines limit this file to be ONLY an executable script. Could it be imagined that the function is this file be called as from this file as a module? If so use if `If __name__ == '__main__':`